pkg/k8s/identitybackend/identity.go

```diff
@@ -367,7 +367,7 @@ func (c *crdBackend) ListAndWatch(ctx context.Context, handler allocator.CacheMu
 				}
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
-				if oldIdentity, ok := newObj.(*v2.CiliumIdentity); ok {
```

Comment:
To avoid indentation levels I would prefer something like this, although fine as is.

```
oldIdentity, ok := oldObj.(*v2.CiliumIdentity)
if !ok { return }

newIdentity, ok := newObj.(*v2.CiliumIdentity)
if !ok { return }

if oldIdentity.DeepEqual(newIdentity) {
	return
}

if id, err := strconv.ParseUint(newIdentity.Name, 10, 64); err == nil {
	handler.OnModify(idpool.ID(id), c.KeyFunc(newIdentity.SecurityLabels))
}
```

Reply:
Hmm, I see your point, but I wouldn't change it in this PR to avoid increasing the diff and hiding the actual change. Ideally, that informer should be also migrated over to `Resource[T]` for consistency with the rest of the code base, and to remove the need for type conversions.